doc/GLM-introduction.rst

```diff
@@ -0,0 +1,435 @@
+===================
+Introduction to GLM
+===================
+
+In Generalized Linear Model (GLM), we mainly want to solve the following problem.
+
+.. math::
+
+    \min_{\beta_0, \beta} \frac{1}{N} \sum_{i = 1}^N w_i \ell (y_i, \beta_0 + \beta^T x_i)
+    + \lambda [0.5(1 - \alpha)\| \beta \|_2^2 + \alpha \| \beta \|_1]
+
+where :math:`\ell (y_i, \beta_0 + \beta^T x_i)` is negative log-likelihood of
+an observation :math:`i`. We explain details of GLM and how we optimize
+the cost function more in introduction to GLM.
+
+
+GLM with elastic net penalty
+----------------------------
+
+**Poisson-like GLM**
+
+The ``pyglmnet`` implementation comes with `poisson`, `poissonexp`, `binomial`,
+`normal` and `multinomial` distributions, but for illustration, I will walk you
+through a particular adaptation of the canonical `Poisson generalized linear model*
+(GLM)`.
+
+.. math::
+
+    \lambda_i = \exp (\beta_0 + \beta^T x_i)
+
+where :math:`x_i \in \mathcal{R}^{p \times 1}, i = \{1, 2, \dots, n\}` are the observed
+independent variables (predictors), :math:`\beta_0 \in \mathcal{R}^{1 \times 1}`,
+:math:`\beta \in \mathcal{R}^{p \times 1}` are linear coefficients.
+:math:`\lambda_i` is also known as the conditional intensity function, conditioned
+on :math:`(\beta_0, \beta)` and :math:`q(z) = \exp(z)` is the nonlinearity.
+
+For numerical reasons, let's adopt a stabilizing non-linearity, known as the
+softplus or the smooth rectifier `Dugas et al., 2001 <http://papers.nips.cc/paper/1920-incorporating-second-order-functional-knowledge-for-better-option-pricing.pdf>`_,
+and adopted by Jonathan Pillow's and Liam Paninski's groups for neural data analysis.
+See for instance: `Park et al., 2014 <http://www.nature.com/neuro/journal/v17/n10/abs/nn.3800.html>`_.
```

Comment:
basically `sphinx-gallery` does the hard work of converting the `.py` file into `.rst` as well as `.ipynb` ...